src/st/no3/46684.c

```diff
@@ -1265,8 +1265,171 @@ INCLUDE_ASM("asm/us/st/no3/nonmatchings/46684", EntityAlucardWaterEffect);
 
 INCLUDE_ASM("asm/us/st/no3/nonmatchings/46684", EntitySplashWater);
 
-// splash animation and sound when Alucard touches water
-INCLUDE_ASM("asm/us/st/no3/nonmatchings/46684", EntitySurfacingWater);
+// ID 0x36
+void EntitySurfacingWater(Entity* self) {
+    Unkstruct8* tileLayout = &g_CurrentRoomTileLayout;
+    Primitive* prim;
+    s16 primIndex;
+    s16 temp_t0;
+    s16 temp_t3;
+    s32 temp_a0;
+    s32 temp_lo;
+    s32 rnd;
+    s32 temp_v1_2;
+    s16 var_s2;
+    s16 temp_a2;
+    u16 params;
+    u16 temp_t2;
+    s32 tempy;
+    s16* ptr;
+    s16 temp;
+    s16 temp3;
+    s16 tempv0;
+    s16 tempa0;
+    s16 x, y;
+    s32 i;
+
+    switch (self->step) {
+    case 0:
+        InitializeEntity(D_80180B00);
```

Comment:
Can we synchronize the data names for these? e.g
```
D_80180B00 -> EntitySurfacingWaterInitData
D_80180A90 -> EntitySurfacingWaterInitData
```

This applies to all the data in these PRs. It will make deduping them easier later.

Reply:
It will make data migration more tedious and difficult, i rather not deduplicate entity functions until we finish the overlay at 100% and migrate the data first.

Reply:
I know what he asked for, D_80180B00 is an Init Chain used not only by this function but 5 others too, i think it's impractical to lose the address in the name and having to look it up at data migration time. All of these Init Chain data entries seem to be at the beginning of the overlay for some reason, they probably are part of the same array or something. I think we are engaging into patterns we don't yet know for sure. I've seen you guys deduplicate entity functions and put them into headers, i don't think that's right at all, the solution needs to be found with data migration in mind, don't get me wrong but i don't think it's time to do this right now, in my opinion is time wasting and will probably make things difficult later.